master/internal/api_tensorboard.go

```diff
@@ -317,6 +321,31 @@ func (a *apiServer) LaunchTensorboard(
 	}, err
 }
 
+func usingCustomImage(req *apiv1.LaunchTensorboardRequest) bool {
+	if req.Config == nil {
+		return false
+	}
+
+	configBytes, err := protojson.Marshal(req.Config)
+	if err != nil {
+		return false
+	}
+
+	type DummyImg struct {
+		Image string `json:"image"`
+	}
+	type DummyEnv struct {
+		Environment DummyImg `json:"environment"`
+	}
```

Comment:
This has a bug, since `environment.image` can be things other than a plain string.

I think you should have a dummy struct containing a `*model.RuntimeItem`, that when `nil` you know it was not set:

```go
type DummyEnv struct {
    Image *model.RuntimeItem `json:"image"`
}
type DummyConfig struct {
    Environment DummyEnv `json:"environment"`
}
```

By using `model.RuntimeItem` you can just borrow the unmarshaling logic, but by using a pointer you can check for nil, which is not normally possible with a `model.Environment` struct.

Also notice that calling the top-level dummy config `DummyEnv` is a bit confusing since it's a dummy replacement for the command config, and what you have as `DummyImg` is really a replacement for the model.Environment struct.